src/Uno.UI.RemoteControl.Host/HotReload/ServerHotReloadProcessor.cs

```diff
@@ -71,30 +74,65 @@ private async Task ProcessConfigureServer(ConfigureServer configureServer)
 				})
 				.ToArray();
 
+			_watcherEventsDisposable = new CompositeDisposable();
+
 			foreach (var watcher in _watchers)
 			{
-				watcher.Changed += OnXamlFileChanged;
+				// Create an observable instead of using the FromEventPattern which
+				// does not register to events properly.
+				// Renames are required for the WriteTemporary->DeleteOriginal->RenameToOriginal that
+				// Visual Studio uses to save files.
+
+				var changes = Observable.Create<string>(o => {
+
+					void changed(object s, FileSystemEventArgs args) => o.OnNext(args.FullPath);
+					void renamed(object s, RenamedEventArgs args) => o.OnNext(args.FullPath);
+
+					watcher.Changed += changed;
+					watcher.Created += changed;
+					watcher.Renamed += renamed;
+
+					return Disposable.Create(() => {
+						watcher.Changed -= changed;
+						watcher.Created -= changed;
+						watcher.Renamed -= renamed;
+					});
+				});
+
+				var disposable = changes
+					.Buffer(TimeSpan.FromMilliseconds(250))
+					.Subscribe(filePaths =>
+					{
+						foreach (var file in filePaths.Distinct().Where(f => Path.GetExtension(f).Equals(".xaml", StringComparison.OrdinalIgnoreCase)))
+						{
+							OnXamlFileChanged(file);
+						}
+					}, e => Console.WriteLine($"Error {e}"));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5553524)